### PR TITLE
Automate contributors list

### PR DESCRIPTION
**Pull request by stefan-kolb**
The list of contributors can now be auto generated via `scripts/generate-authors.sh`.

It uses an additional `.mailmap` file to merge different name/email configurations of unique authors.
The list is in sync with the old manual list from `About.html`.
Missing manual entries are concatenated in `scripts/generate-authors.sh`for now.

Imho, we should either skip missing authors, or map them to unmapped commit emails.
I'm not sure how many commits are missing in our history and in which way these people contributed.
Maybe @koppor can shed some light on this.

The idea is to use this file to replace our contributors list inside `About.html` automatically without losing contributor's information or being inconsistent between languages etc.
We may have to do some changes to the file structure, e.g. attribute _name_ only, but this PR is mainly for unifying our data first.

The script itself `scripts/generate-authors.sh` can either be included in the build process or executed manually at times, as it may be beneficial if we oversee the mapping process by a human.

### CODE REVIEW

File: scripts/generate-authors.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	Mike Smoot <>
+	Ulrich Stärk <>
+	Martin Stolle <>
+	David Weitzman <>
+	John Zedlewski <>
+	Samin Muhammad Ridwanul Karim <>
+	Stefan Robert <>
+	EOF
+
+	echo -e "$authors\n$(git log --format='%aN <%aE>')" | LC_ALL=C.UTF-8 sort -uf
```
Comment (simonharrer) [inline scripts/generate-authors.sh]: Can you please use the full options for sort -uf, and place a comment what LC_ALL=C.UTF-8 is for. 
Maybe also explain the format string used in the log statement.

**Comment by koppor** [comment]
@stefan-kolb The thing is that the old version control system used before was Subversion, which does not support the difference between author and committer. [JabRef switched to git in November, 2011](http://osdir.com/ml/java-jabref-devel/2011-11/msg00006.html). That means, all commits before November 2011 may have wrong author information. Furthermore, after November 2011 not all patches have been applied using `--author`.

**Comment by ThomasA** [comment]
Also, is the author list generation script automatically part of JabRef's build procedure? In that case, it could exclude most Windows users?

**Comment by stefan-kolb** [comment]
@ThomasA It is not (yet?) part of the build procedure. Even if it was Windows users should be able to run it with Cygwin or MSYS.

**Comment by ThomasA** [comment]
I was just thinking that if this is the only shell-dependent step in 
building JabRef, it might be a major nuisance to anyone on Windows if 
they would have to install and build in Cygwin just for this. This could 
potentially disrupt the workflow of most Windows contributors.
I am just speculating here. I don't know how many Windows contributors 
there are and I am using Linux myself.

**Comment by simonharrer** [comment]
I do not think that this is an issue because when having git installed locally on windows, there must at the moment be also the basic unix tools available as well because git relies upon them. (I am a windows developer)

But this stirrs up another interesting question. The previous scripts that were used to generate the homepage pages out of the authors were written in perl, which is yet another dependency. And we have python scripts in the repository as well. Maybe we should just use a single scripting language for all the scripts to reduce the learning overhead?

**Comment by ThomasA** [comment]
OK, I just wanted to make sure this issue was considered.
As for a unifying scripting language, I highly favour Python but then 
again, I hardly contribute to JabRef nowadays.

**Comment by mlep** [comment]
Of course, using a single scripting language will make the maintenance easier.
I suggest to define a preferred scripting language for *future development*.

About existing scripts: I believe to rewrite all of them is not a high priority for JabRef development. Maybe conversion of the critical ones (which are to be identified).

**Comment by stefan-kolb** [comment]
Ok, I think we are ready to merge this. Any objections?

**Pull request merged**